src/mono/CMakeLists.txt

```diff
@@ -326,6 +321,10 @@ elseif(TARGET_SYSTEM_NAME STREQUAL "Emscripten")
   endif()
 elseif(TARGET_SYSTEM_NAME STREQUAL "WASI")
   set(TARGET_WASI 1)
+  set(DISABLE_THREADS 1)
```

Comment:
should we pass this in via the .proj file so we can control it via the `$(MonoWasmThreads)` msbuild property like we do for browser?